packages/backend/controllers/clubs.js

```diff
@@ -1,38 +1,88 @@
-const admin = require('firebase-admin');
-const { ClubSchema } = require('@clubhub/common/models');
-
-
-async function addClub(data){
-    console.log({body: data})
-    let result = ClubSchema.validate(data, { stripUnknown: { objects: true } }) //strip removes unwanted attribute in club
-    console.log(result);
-    
-    const db = admin.firestore();
-    if (result.error){
-        throw Error(result.error.details.messages)
-    }else{
-        let newClub =  await db.collection('clubs').add({
-            name: data.name,
-            active: data.active,    //three attributes that are required in schema
-            description: data.description
-        })
-        return newClub; //in case wanted to do stuff after
-    }
+const admin = require("firebase-admin");
+const { ClubSchema } = require("@clubhub/common/models");
+
+async function addClub(data) {
+  console.log({ body: data });
+  let result = ClubSchema.validate(data, {
+    stripUnknown: { objects: true },
+    abortEarly: false
+  }); //strip removes unwanted attribute in club
+  console.log(result);
+
+  const db = admin.firestore();
+  if (result.error) {
+    throw Error(result.error.details.map(item => item.message).join(", ")); //joins errors, strips result.error of unwanted clutter
+  } else {
+    let newClub = await db.collection("clubs").add({
+      name: data.name,
+      active: data.active, //three attributes that are required in schema
+      description: data.description
+    });
+    return newClub; //in case wanted to do stuff after
+  }
 }
 
-async function getAllClubs(){
-    
-    const db = admin.firestore();
-    const clubData = await db.collection('clubs').get();
-    let allClubs = []; //init list to store clubData
-    clubData.forEach((club) =>{
-        allClubs.push({...club.data(), id: club.id}); //place id into collection of club data
+async function editClub(club_id, data) {
+  let result = ClubSchema.validate(data, {
+    //editClub assumes that frontend will always update a club by sending a request of ALL properties
+    stripUnknown: { objects: true },
```

Comment:
should this line be `stripUnknown: true`?

Reply:
Im not sure, I just followed the documentation

Reply:
I believe it has to be `stripUnknown: true` similar to `abortEarly: false`

Reply:
Ok, Ill change it now